src/libraries/System.Text.Json/Common/JsonNumberHandling.cs

```diff
@@ -3,6 +3,16 @@
 
 namespace System.Text.Json.Serialization
 {
+#if BUILDING_SOURCE_GENERATOR
+    // cref references below are not present when compiling this assembly.
+    internal enum JsonNumberHandling
```

Comment:
<details>
<summary>
We could, but we'd need more #ifdefs. (Click to see what it'd look like)
</summary>

```cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace System.Text.Json.Serialization
{
#if !BUILDING_SOURCE_GENERATOR
    /// <summary>
    /// Determines how <see cref="JsonSerializer"/> handles numbers when serializing and deserializing.
    /// </summary>
#endif
    [Flags]
#if BUILDING_SOURCE_GENERATOR
    internal
#else
    public
#endif
    enum JsonNumberHandling
    {
#if !BUILDING_SOURCE_GENERATOR
        /// <summary>
        /// Numbers will only be read from <see cref="JsonTokenType.Number"/> tokens and will only be written as JSON numbers (without quotes).
        /// </summary>
#endif
        Strict = 0x0,

#if !BUILDING_SOURCE_GENERATOR
        /// <summary>
        /// Numbers can be read from <see cref="JsonTokenType.String"/> tokens.
        /// Does not prevent numbers from being read from <see cref="JsonTokenType.Number"/> token.
        /// Strings that have escaped characters will be unescaped before reading.
        /// Leading or trailing trivia within the string token, including whitespace, is not allowed.
        /// </summary>
#endif
        AllowReadingFromString = 0x1,

#if !BUILDING_SOURCE_GENERATOR
        /// <summary>
        /// Numbers will be written as JSON strings (with quotes), not as JSON numbers.
        /// </summary>
#endif
        WriteAsString = 0x2,

#if !BUILDING_SOURCE_GENERATOR
        /// <summary>
        /// The "NaN", "Infinity", and "-Infinity" <see cref="JsonTokenType.String"/> tokens can be read as
        /// floating-point constants, and the <see cref="float"/> and <see cref="double"/> values for these
        /// constants (such as <see cref="float.PositiveInfinity"/> and <see cref="double.NaN"/>)
        /// will be written as their corresponding JSON string representations.
        /// Strings that have escaped characters will be unescaped before reading.
        /// Leading or trailing trivia within the string token, including whitespace, is not allowed.
        /// </summary>
#endif
        AllowNamedFloatingPointLiterals = 0x4
    }
}
```
</details>

I'm avoiding the crefs since the referenced STJ types/members are not present in netstandard2.0 (source generator TFM)

Reply:
Can you instead just disable XML doc comment generation for the source generator?  We shouldn't be publishing that anyways.

Reply:
Done. Suppressing CS1574 project-wide made the errors go away. So now we have the same definition, but now just `internal` in the source generator.